src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -0,0 +1,143 @@
+    super(element, MissingTerms.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, MissingTerms, options);
+    this.addMissingTerms();
+  }
+  /**
+   *Returns all original basic query expression keywords that were not matched by the result item the component instance is associated with.
+   */
+  public get missingTerms(): string[] {
+    let cleanMissingTerms: string[] = [];
+    each(this.result.absentTerms, (term: string) => {
+      const regex = new RegExp(`${this.wordBoundary}${term}${this.wordBoundary}`);
+      if (regex.test(this.result.state.q)) {
+        cleanMissingTerms.push(term);
+      }
+    });
+    return cleanMissingTerms;
+  }
+  /**
+   * @param term : The `string` to be re-injected in the query as exact match
+   *
+   * Re-injects a term as an exact phrase match expression in the query.
+   */
+  public includeTermInQuery(term: string) {
+    if (this.missingTerms.indexOf(term) === -1) {
+      return;
+    }
+
+    let newQuery: string = this.queryStateModel.get('q');
+    const regex = new RegExp(`${this.wordBoundary}${term}${this.wordBoundary}`, 'g');
+    let stillResults = true;
+    let results;
+    while (stillResults) {
+      results = regex.exec(newQuery);
+      stillResults = results !== null;
+      if (stillResults) {
+        newQuery = [newQuery.slice(0, ++results.index), '"', term, '"', newQuery.slice(results.index + term.length)].join('');
+      }
+    }
+    this.queryStateModel.set('q', newQuery);
+  }
+
+  private addMissingTerms() {
+    if (this.missingTerms.length === 0) {
+      return;
+    }
+    $$(this.element).append(this.buildContainer().el);
+  }
+
+  private buildContainer(): Dom {
+    const resultCell = $$('div', { className: 'coveo-result-cell' }, this.buildCaption());
+    this.buildMissingTerms().forEach(term => {
+      resultCell.append(term.el);
+    });
+
+    return $$('div', { className: 'coveo-result-row' }, resultCell);
+  }
+
+  private buildCaption(): Dom {
+    return $$('span', { className: 'coveo-field-caption' }, this.options.caption);
+  }
+
+  private buildMissingTerms(): Dom[] {
+    const terms: Dom[] = this.missingTerms.map(term => {
+      return this.clickableButtonIfEnable(term);
+    });
+    return terms;
+  }
+
+  private executeNewQuery(missingTerm: string) {
+    this.usageAnalytics.logSearchEvent<IAnalyticsIncludeMissingTerm>(analyticsActionCauseList.missingTermsSearch, {
+      missingTerm: missingTerm
+    });
+    this.queryController.executeQuery();
+  }
+
+  private clickableButtonIfEnable(term: string): Dom {
+    if (this.options.clickable) {
+      const termElement = $$('button', { className: 'coveo-missing-term' }, term);
+      termElement.on('click', () => {
+        this.includeTermInQuery(term);
+        this.executeNewQuery(term);
+      });
+      termElement.addClass('clickable');
```

Comment:
You can add this class directly when you create the element : 

```
 $$('button', { className: 'coveo-missing-term coveo-clickable' }, term);
```

Also, you need to always prefix the classname with `coveo-` to prevent conflict when we integrate in external system/websites.